### PR TITLE
INTMDB-153: Create 0.8.0-upgrade-guide.html.markdown

### PR DESCRIPTION
**Pull request by themantissa**
## Description

Guide for IAM creds/roles switch needs and private link deprecation

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ x] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the Terraform contribution guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code

### CODE REVIEW

**Review by leofigy — APPROVED** [review]
LGTM, Hi @coderGo93 , could you please check as well.

**Review by coderGo93 — APPROVED** [review]
LGTM!

File: website/docs/guides/0.8.0-upgrade-guide.html.markdown [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+### Step 1: Configuring Cloud Provider Access
+
+If you programmatically create new Atlas projects and use AWS KMS encryption keys for encryption at rest or you are ready to switch an existing Atlas project using AWS KMS encryption keys for encryption at rest to using IAM users to roles, you will need to first configure the [cloud provider access](https://docs.atlas.mongodb.com/security/set-up-unified-aws-access) with https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_provider_access.  The Atlas documentation for [cloud provider access](https://docs.atlas.mongodb.com/security/set-up-unified-aws-access) will walk you through the process in general.  Configuring cloud provider access with Terraform is a multi step process: configure and apply the Atlas cloud_provide_access resource, modify your AWS IAM role trust policy with AWS, then add the IAM Assumed Role ARN to your Atlas resource definition and apply again to authorize. After this process you can then configure or update encryption at rest for your project.  For example:
```
Comment (shum) [inline website/docs/guides/0.8.0-upgrade-guide.html.markdown]: Might recommend breaking up the steps into a numbered list for readabilty, eg:
```
Configuring cloud provider access with Terraform is a multi step process:
1. Configure and apply the Atlas `cloud_provider_access` resource // note there was a typo here for "provider"
2. Modify your AWS IAM role trust policy with AWS
3. Add the IAM Assumed Role ARN to your Atlas resource definition
4. Apply again to authorize.
```

File: website/docs/guides/0.8.0-upgrade-guide.html.markdown [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+**Example 1st Terraform Apply**
```
Comment (shum) [inline website/docs/guides/0.8.0-upgrade-guide.html.markdown]: This doesn't seem to be rendering properly - Github is just squashing all the newlines and making this section hard to read. (Might be Github thing) Consider adding ` ``` ` around the section?

File: website/docs/guides/0.8.0-upgrade-guide.html.markdown [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    }
+    #Information you’ll need to edit your AWS policy 
+        output "cloud_provider_access_external_id" {
+        value = mongodbatlas_cloud_provider_access.test.atlas_assumed_role_external_id
+    }
+    #Information you’ll need to edit your AWS policy 
+        output "cloud_provider_access_arn" {
+        value = mongodbatlas_cloud_provider_access.test.atlas_aws_account_arn
+    }
+    #Information you’ll need to enable encryption at rest later
```
Comment (shum) [inline website/docs/guides/0.8.0-upgrade-guide.html.markdown]:
```suggestion
    #Information you’ll need to enable BYOK encryption at rest later
```

File: website/docs/guides/0.8.0-upgrade-guide.html.markdown [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
```
Comment (shum) [inline website/docs/guides/0.8.0-upgrade-guide.html.markdown]:
```suggestion
    #Information you’ll need to enable BYOK encryption at rest later
```

File: website/docs/guides/0.8.0-upgrade-guide.html.markdown [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        output "cloud_provider_role_id" {
+        value = mongodbatlas_cloud_provider_access.test.role_id
+    }
+
+**Example 2nd Apply**
+    resource "mongodbatlas_cloud_provider_access" "test" {
```
Comment (shum) [inline website/docs/guides/0.8.0-upgrade-guide.html.markdown:57]: Same as above re preformatted text.

File: website/docs/guides/0.8.0-upgrade-guide.html.markdown [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+**If the project exists** already and is using Customer Key Management with AWS KMS then you can update the encryption_at_rest Terraform resource to use the new IAM role: https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/encryption_at_rest
+
+**IMPORTANT!!!  If you switch your Atlas project from credentials-based access (IAM user) to role-based access (IAM role) to your encryption keys, you cannot undo the role-based access configuration and revert to credentials-based access for that project.**
+
+For example (these are example values, replace with your values) within the mongodbatlas_encryption_at_rest resource:
+
+Existing:
+    aws_kms = {
```
Comment (shum) [inline website/docs/guides/0.8.0-upgrade-guide.html.markdown:100]: Same as above re preformatted text.

File: website/docs/guides/0.8.0-upgrade-guide.html.markdown [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        region                             = "US_EAST_1"
+      }
+
+New:
+    aws_kms = {
+        enabled                               = true
+        role_id                          = "5f232b94af0a6b41747akx2d"
+        customer_master_key_id         = "030gce02-586d-48d2-a999-05ea954fdd0g"
+        region                             = "US_EAST_1"
+      }
```
Comment (shum) [inline website/docs/guides/0.8.0-upgrade-guide.html.markdown:117]: Same as above re preformatted text.

**Review by shum — CHANGES_REQUESTED** [review]
Minor changes suggested! Looks good!

File: website/docs/guides/0.8.0-upgrade-guide.html.markdown [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+If you programmatically create new Atlas projects and use AWS KMS encryption keys for encryption at rest or you are ready to switch an existing Atlas project using AWS KMS encryption keys for encryption at rest to using IAM users to roles, you will need to first configure the [cloud provider access](https://docs.atlas.mongodb.com/security/set-up-unified-aws-access) with https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_provider_access.  The Atlas documentation for [cloud provider access](https://docs.atlas.mongodb.com/security/set-up-unified-aws-access) will walk you through the process in general.  
+
+Configuring cloud provider access with Terraform is a multi step process:
+
+1) Configure and apply the Atlas cloud_provide_access resource
+2) modify your AWS IAM role trust policy with AWS
+3) Add the IAM Assumed Role ARN to your Atlas resource definition
+4) Apply again to authorize.
+After this process you can then configure or update encryption at rest for your project.  For example:
```
Comment (shum) [inline website/docs/guides/0.8.0-upgrade-guide.html.markdown:29]: This line's showing up weird (gets swallowed into the list). Maybe add one extra new line after the list.

File: website/docs/guides/0.8.0-upgrade-guide.html.markdown [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+
+In order to transition from the deprecated resources to the new without disabling private link you will need to:
+
+1) [Remove](https://www.terraform.io/docs/commands/state/rm.html) any currently managed private_endpoint and private_endpoint_interface_link resources from your Terraform state.  We suggest saving the state information before making any changes both in case of mistake and to use for the next steps.
+2) Create a new resource in your Terraform configuration file using the documentation as a guide: [privatelink_endpoint](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/privatelink_endpoint) and [privatelink_endpoint_service](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/privatelink_endpoint_service)
+3) Then [import](https://www.terraform.io/docs/commands/import.html) the privatelink information into the new resources, e.g: 
+terraform import mongodbatlas_privatelink_endpoint.test {project_id}-{private_link_id}-{provider_name}
+Mongodbatlas_privatelink_endpoint_service.test {project_id}--{private_link_id}--{endpoint_service_id}--{provider_name}
```
Comment (shum) [inline website/docs/guides/0.8.0-upgrade-guide.html.markdown]:
```suggestion
terraform import mongodbatlas_privatelink_endpoint.test {project_id}-{private_link_id}-{provider_name}
Mongodbatlas_privatelink_endpoint_service.test {project_id}--{private_link_id}--{endpoint_service_id}--{provider_name}
```
hah can't add additional ` ``` `s but we should probably wrap this block too.

**Review by shum — APPROVED** [review]
tiny things. LGTM!!

**Pull request merged**